tests/test_interface.py

```diff
@@ -4,135 +4,101 @@
 
```

Comment:
Ok, I still do not see any simple test for groups behavior in one place aka. `container`.
But this could be a next task as you wish. So we should have your solution tested on some experiment.